src/core/ConfigManager.cpp

```diff
@@ -139,7 +139,8 @@ void ConfigManager::upgrade()
 	}
 
 	// Runs all necessary upgrade methods
-	std::for_each( UPGRADE_METHODS.begin() + m_configVersion, UPGRADE_METHODS.end(),
+	std::size_t max = std::min(static_cast<std::size_t>(m_configVersion), UPGRADE_METHODS.size());
```

Comment:
The variable name looks very odd to me. If I were fixing this bug, I would use a `if` clause instead, but that's just a matter of taste.